### PR TITLE
gcc-7: fix sparc build

### PR DESCRIPTION
**Pull request by tsoome**

### CODE REVIEW

**Review by AndWac — APPROVED** [review]

File: components/developer/gcc-7/Makefile [hunk 1]
```diff
@@ -46,7 +46,7 @@ include $(WS_MAKE_RULES)/gcc-component.mk
 
 CONFIGURE_OPTIONS+= --enable-plugins
 CONFIGURE_OPTIONS+= --enable-objc-gc
-CONFIGURE_OPTIONS+= --enable-initfini-array
+CONFIGURE_OPTIONS.i386+= --enable-initfini-array
```
Comment (pyhalov) [inline components/developer/gcc-7/Makefile:49]: This doesn't work as expected

File: components/developer/gcc-7/gcc-7.p5m [hunk 2]
```diff
@@ -1088,20 +1088,21 @@ file path=usr/gcc/7/lib/gcc/$(GNU_TRIPLET)/$(COMPONENT_VERSION)/include/stdfix.h
...
-file path=usr/gcc/7/lib/gcc/$(GNU_TRIPLET)/$(COMPONENT_VERSION)/include/xsaveintrin.h
-file path=usr/gcc/7/lib/gcc/$(GNU_TRIPLET)/$(COMPONENT_VERSION)/include/xsaveoptintrin.h
-file path=usr/gcc/7/lib/gcc/$(GNU_TRIPLET)/$(COMPONENT_VERSION)/include/xsavesintrin.h
-file path=usr/gcc/7/lib/gcc/$(GNU_TRIPLET)/$(COMPONENT_VERSION)/include/xtestintrin.h
+file path=usr/gcc/7/lib/gcc/$(GNU_TRIPLET)/$(COMPONENT_VERSION)/include/visintrin.h variant.arch=sparc
+file path=usr/gcc/7/lib/gcc/$(GNU_TRIPLET)/$(COMPONENT_VERSION)/include/wmmintrin.h variant.arch=i386
+file path=usr/gcc/7/lib/gcc/$(GNU_TRIPLET)/$(COMPONENT_VERSION)/include/x86intrin.h variant.arch=i386
+file path=usr/gcc/7/lib/gcc/$(GNU_TRIPLET)/$(COMPONENT_VERSION)/include/xmmintrin.h variant.arch=i386
+file path=usr/gcc/7/lib/gcc/$(GNU_TRIPLET)/$(COMPONENT_VERSION)/include/xopintrin.h variant.arch=i386
+file path=usr/gcc/7/lib/gcc/$(GNU_TRIPLET)/$(COMPONENT_VERSION)/include/xsavecintrin.h variant.arch=i386 variant.arch=i386
```
Comment (pyhalov) [inline components/developer/gcc-7/gcc-7.p5m]: two variant.arch=i386

**Review by pyhalov — CHANGES_REQUESTED** [review]

File: components/developer/gcc-7/Makefile (hunk 1 above)
Comment (tsoome) [reply components/developer/gcc-7/Makefile:49]: no?

make-rules/gcc-component.mk:CONFIGURE_OPTIONS+= $(CONFIGURE_OPTIONS.$(MACH))

Comment (pyhalov) [reply components/developer/gcc-7/Makefile:49]: Sorry, was wrong.

**Review by pyhalov — APPROVED** [review]

**Comment by pyhalov** [comment]
Built illumos-gate with this gcc for testing.

**Pull request merged**